transform/_procesador_i3.py

```
@@ -0,0 +1,455 @@
+                how='left',
+                suffixes=('_current', '_previous')
+            )
+            merged['volumenes_previous'] = merged['volumenes_previous'].fillna(0)
+            merged['volumenes_diff'] = merged['volumenes_current'] - merged['volumenes_previous']
+            
+            session_result = merged.rename(columns={'volumenes_diff': 'volumenes'})
+            session_result = session_result[['datetime_utc', 'tecnologia', 'volumenes']]
+            session_result['id_mercado'] = session_id
+            return session_result
+        except Exception as e:
+            raise ValueError(f"Error calculating session differences: {e}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)